### DISCUSSION TITLE
Clone/Copy an OffLatticeSimulation that is loaded from file rather than loading over and over

### DISCUSSION THREAD

**Discussion by Elphikaus** [Chaste/Chaste#156] [Q&A] [answered]
As the title says I am interested in cloning or copying an object.

I run a small simulation and save it to file using the archiving tool.

Then I load the simulator from file and run an extended simulation dependent on parameter values.

Each iteration I upload the simulator from file again.

I wonder if this is taking up a lot of resources as each of my runs takes about 18 mins.

Would using a copy constructor save time?

If so, how do I go about doing that?

**Comment by mirams** [comment, marked as answer]
Hi, I don't think a copy constructor exists for cell-based Simulation objects, partly because there's a concept of there only being one Simulation Time in existence globally and things like that (not a different one for each Simulation object if you had more than one), so chaos may ensue if more than one existed at once. Resetting simulation time and a few other things is specially handled by the archiving classes.

It would be worth timing (see the `Timer` class) to see how much time is actually spent loading from the archive.

One strategy that I have used to speed that up is having a binary archive rather than an ascii one. Here are some example timings from the cardiac side for that, so you can potentially halve the loading time just by using a binary archive:

| Archive type | size (MB) | load time (secs) |
|---|---|---|
| standard ascii | 820| 31.8 |
| standard binary | 620| **14.6** |
| compressed ascii | 181| 39.1 |
| compressed binary | **133**| 17.3 |

To use a binary archive is just a one-line change in classes you are archiving yourself - see for example [`TestArchiving.hpp`](https://github.com/Chaste/Chaste/blob/f959aeba7f54188bd5c308b38c3c615a1379b882/global/test/TestArchiving.hpp#L391C36-L391C36), I guess here you might want to alter lines 108-109 and 132-133 in `cell_based/src/simulation/CellBasedSimulationArchiver.hpp` to do that with all your archived cell based simulations.
